### PR TITLE
[SPARK-35378][SQL][FOLLOWUP] Restore the command execution name for DataFrameWriterV2

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a followup of https://github.com/apache/spark/pull/32513

It's hard to keep the command execution name for `DataFrameWriter`, as the command logical plan is a bit messy (DS v1, file source and hive and different command logical plans) and sometimes it's hard to distinguish "insert" and "save".

However, `DataFrameWriterV2` only produce v2 commands which are pretty clean. It's easy to keep the command execution name for them.

### Why are the changes needed?

less breaking changes.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @beliefer

File: sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala [hunk 1]
```diff
@@ -778,8 +778,7 @@ class WriteDistributionAndOrderingSuite
 
     sparkContext.listenerBus.waitUntilEmpty()
 
-    assert(executedPlan.isInstanceOf[CommandResultExec])
-    executedPlan.asInstanceOf[CommandResultExec].commandPhysicalPlan match {
+    executedPlan match {
```
Comment (beliefer) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala:781]: Why we need this change?

Comment (beliefer) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala:781]: Oh. I got it.

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/WriteDistributionAndOrderingSuite.scala:781]: I fixed https://github.com/apache/spark/pull/32919/files#diff-23a90192be7d7c83780c61292bc061b2eddb79de6cd94d8234dddce781f60c1eL195

So I reverted these changes from your previous commit.

**Review by beliefer — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks for review, merging to master!

**Pull request closed without merging**